languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -34110,6 +34110,125 @@ though, and has since been slightly modified)-->
+              <suggestion>\3,</suggestion>
+              <example correction="effet,">En <marker>effet</marker> il ne le remarqua pas.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <token>par</token>
+                  <marker>
+                      <token regexp="yes">conséquent|contre</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\3,</suggestion>
+              <example correction="contre,">Par <marker>contre</marker> il ne le remarqua pas.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token regexp="yes">moi|toi</token>
+                  <token regexp="yes">aussi|non|et|qui|absolu|ou</token>
+              </antipattern>
+              <antipattern>
+                  <token regexp="yes">moi|toi</token>
+                  <token>d'</token>
+                  <token>abord</token>
+              </antipattern>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <marker>
+                      <token regexp="yes">moi|toi</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\2,</suggestion>
+              <example correction="Moi,"><marker>Moi</marker> je ne le sais pas.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <marker>
+                      <token regexp="yes">lui|elle|eux|nous|vous|elles</token>
+                  </marker>
+                  <token postag="R.* pers.* suj.*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\2,</suggestion>
+              <example correction="Lui,"><marker>Lui</marker> il ne le sais pas.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token postag="V.*" postag_regexp="yes"/>
+                  <token postag="R.* pers.* suj.*" postag_regexp="yes"/>
+                  <token>que</token>
+                  <token postag="R.*" postag_regexp="yes"/>
+                  <token postag="V.*" postag_regexp="yes" skip="-1"/>
+              </antipattern>
+              <antipattern>
+                  <token>qu'</token>
+                  <token>elle</token>
+                  <token>que</token>
+              </antipattern>
+              <pattern>
+                  <marker>
+                      <token postag="R.* pers.* suj.*" postag_regexp="yes"/>
+                      <token>que</token>
```

Comment:
would this work for "qui" as well? "Elle qui a beacoup voyagé"?

Reply:
After some thorough research, I think this rule should not be in this rulegroup. Indeed, this rule is very tricky. In French there is a comma if the part after "qui/que" gives an explanation that is not necessary in the sentence. The comma is forbidden if the part after "qui/que" is needed in the sentence. The difference is very subtle and depends on the whole context.